modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -103,6 +141,17 @@ private boolean doBinaryComparisonCoercion(SqlCallBinding binding) {
         RelDataType leftType = binding.getOperandType(0);
         RelDataType rightType = binding.getOperandType(1);
 
+        if (SqlTypeUtil.isIntType(leftType) && SqlTypeUtil.isCharacter(rightType)
```

Comment:
Could you please write a comment explaining a problem does this code solves.

Reply:
remove it